### ISSUE TITLE
iloc[-7] with 4h timeframe after bot start

### ISSUE THREAD

**Issue by myNextCoder** [freqtrade/freqtrade#9896]
## Describe your environment

  * Operating system: Windows Server 2022
  * Python Version: Current Miniconda (`python -V`)
  * CCXT version: _____ (`pip freeze | grep ccxt`)
  * Freqtrade Version: 2023.12 (`freqtrade -V` or `docker compose run --rm freqtrade -V` for Freqtrade running in docker)
  
## Your question

If I am not wrong, Freqtrade would need up to 1 day, to get the values of (for example) ATR for the last 6 4h candles. As I print them out on a fresh started bot, the values for iloc[-1] to iloc[-7] are the same. Doing it on the 1m timeframe, the values differ. So it seems, that via the ta lib, ta.ATR might not have access to these 6 past 4h candles, if the bot did not run at that specific time. For this reason, startup_candle_count is important. Tell me, if you see an approach, to access those old candles, before this warmup time (dry and live, not backtest).

*Ask the question you have not been able to find an answer in the [Documentation](https://www.freqtrade.io/en/latest/)*

**Comment by stash86** [comment]
Assuming your main timeframe is 4h, then no, the data should already available on startup. But if 4h is your informative, then it might not be available, depending on your main timeframe, because of how they are merged

https://botacademy.ddns.net/2023/07/27/how-dataframe-looked-like-after-merging-informative-data/

In the end, we only store specified amounts of rows based of the main timeframe. Any informative timeframes data that are way past the main timeframe's last timestamp gonna be cut off

**Comment by myNextCoder** [comment]
I thank you. Is there a way, to access old data during dry- and live-run like with backtest, by setting a historical data folder, which Freqtrade uses, if online data is not yet available? On the page "start the bot" I see the command line option, to include the data path, but will it be used, as a workaround for informative pairs, to analyze old candles?

**Comment by stash86** [comment]
No you can't

**Comment by myNextCoder** [comment]
Ok, I see. One last question, to handle the warming up time: Does  startup_candle_count resets and starts to count again from 0 every time, as I push /reload_config?

**Comment by xmatthias** [comment]
there is no "warm up" time (not unless you're using special functionality of freqAI, really).
Either the candles are within the lookbehind window - or they are not. leaving a bot running for 4 hours will not change this - not unless you implement this caching manually within the exchange.

The most likely cause is that your _main timeframe_ won't get you 8x 4h back (e.g. on 1m candles) - so a merge will drop candles up to the starting point of the initial candles.

if 4h is your main timeframe - then this won't happen (if it does, then it's either a brandnew pair, or the exchange had a prolonged downtime (of several days ...).


Without a proper example on how you're using which dataframe (informative, main, ...) - it's however difficult to help you further.

**Comment by myNextCoder** [comment]
I thank you very much. I may post code for you in a new ticket later. Currently it is very long, I need to extract important spots first, which still make sense, and are understandable.
At least I was already able, - my default timeframe is 1m - to see the print out of my ATR values (discussed above), as I switched from my initial 4h informative timeframe to 30m: Here, first the values have been 0, changing to meaningful values, after half an hour, as expected.
If I don't get the desired result with the startup candles, I let count a timer, and enter a trade, only after a specific time.

**Comment by xmatthias** [comment]
Well your description already tells a lot.

You're working on 1m candles - let's assume on binance - so you get 1000 candles by default - or roughly 4 4h candles (1000 candles is roughly 16 hours).

Going back 7 4h candles (a bit more than 1 day) with this combination will therefore not really work - as these candles are not mergable - and will therefore "fall out" the back of the lookback window.

the same applies for the "first" values - they might be 0 - or might be the real candle - depending on the the actual first 1m candle you get - which will shift over time.

It's all a question of "which timestamps are in both dataframes", really. If the informative pair doesn't have commonalities - rows will be dropped (e.g. you cannot merge 1w candles onto 1m candles - it'll only have content once a week).

---

The assumption that "waiting" for a specific amount of candles will resolve either of these issues is based on wrong understanding.
Dates may align eventually - but they'll missalign again after a few candles - so you'll again start with empty initial candles.


As an aside - using `.iloc[-7]` is a lookahead bias in backtesting unless it's only used in callbacks - which if i'm honest, your question doesn't sound like that's the case.
